unstable/wlr-layer-shell-unstable-v1.xml

```diff
@@ -203,21 +204,85 @@
       <arg name="left" type="int"/>
     </request>
 
+    <enum name="keyboard_interactivity" since="4">
+      <description summary="types of keyboard interaction possible for a layer shell surface">
+        Types of keyboard interaction possible for layer shell surfaces. The
+        rationale for this is twofold: (1) some applications are not interested
+        in keyboard events and not allowing them to be focused can improve the
+        desktop experience; (2) some applications will want to take exclusive
+        keyboard focus.
+      </description>
+
+      <entry name="no_focus" value="0">
+        <description summary="no keyboard focus is possible">
+          This value indicates that this surface is not interested in keyboard
+          events and the compositor should never assign it the keyboard focus.
+
+          This is the default value, set for newly created layer shell surfaces.
+
+          This is useful for e.g. desktop widgets that display information or
+          only have interaction with non-keyboard input devices.
+        </description>
+      </entry>
+      <entry name="exclusive" value="1">
+        <description summary="request exclusive keyboard focus">
+          Request exclusive keyboard focus if this surface is above the shell surface layer.
+
+          For layers above the shell surface layers, the seat will always give
```

Comment:
This sounds a little bit like this is writing down compositor policy. I'd like to make it clear that compositors can decide to implement a different policy, e.g. don't give focus to any keyboard-interactive layer surface when a lockscreen is active.

Reply:
Ah, it seems like the wording comes from the original description for `set_keyboard_interactivity`.

Reply:
Yes, for this option, I took the current wording :)

I agree there is no guarantee to e.g. prevent stealing the focus from a lock screen either in the current protocol or this PR. I feel this could be a separate issue though.

Reply:
Yes, you're right. We can fix it separately.